### ISSUE TITLE
nifti2dicom conversion no error but no output

### ISSUE THREAD

**Issue by MPardey** [ENHANCE-PET/nifti2dicom#33]
Hey,
I'm trying to run the following in terminal. I'm in my subject folder, and all of my paths are correct.
nifti2dicom -d ./DICOM/PA000001/ST000001/SE000001 -n ./bids/derivatives/sub-iai343y_ses-20241021_run-01_acq-qsm_Chimap.nii -o ./QSM-dicomconvert -desc "QSM Analysis" -t img

When it starts I get the nifti2dicom logo, it thinks for a bit, and then returns to the command line prompt without producing any errors. However, when I check my output folder, it is empty.

Please note I am a coding novice.

Cheers
Marg

**Comment by Keyn34** [comment]
Hi, are you using relative paths, or is the . in front of the paths to hide private paths? 

Can you run the software with the complete system path, please?

**Comment by LalithShiyam** [comment]
I think it's the path as well. Please provide the entire path rather than relative path, for now. we will provide a fix soon-ish.

**Comment by MPardey** [comment]
Morning,
Thanks for the feedback. Following your message I thought it might be an issue with Linux reading from a Windows folder, so used data within the Linux system. From the root folder I ran the below:

(base) jovyan@neurodesktop-2024-10-22:~$ ls
Desktop  MargQSMplay.txt  conda-readme.md  dicoms-sorted  example-notebooks  funtional_imaging  neurodesktop-storage  thinclient_drives  untitled.txt  work
(base) jovyan@neurodesktop-2024-10-22:~$ nifti2dicom -d example-notebooks/books/041/DICOM/PA000001/ST000001/SE000001/ -n example-notebooks/books/041/bids/derivatives/qsmxt-2024-11-19-235919/sub-041/ses-20241101/anat/sub-041_ses-20241101_run-01_acq-qsm_Chimap.nii -o example-notebooks/books/041/QSM-dicomconvert/ -desc "QSM Analysis" -t img
 
          _ ______  _ ___       ___                    
   ____  (_) __/ /_(_)__ \ ____/ (_)________  ____ ___ 
  / __ \/ / /_/ __/ /__/ // __  / / ___/ __ \/ __ `__ \
 / / / / / __/ /_/ // __// /_/ / / /__/ /_/ / / / / / /
/_/ /_/_/_/  \__/_//____/\__,_/_/\___/\____/_/ /_/ /_/
A package to convert NIfTI images to DICOM format using a reference DICOM series. Nifti2dicom is a part of the ENHANCE.PET (https://enhance.pet) framework
 
(base) jovyan@neurodesktop-2024-10-22:~$ 

Still no output and no errors.
Appreciate your assistance.

**Comment by LalithShiyam** [comment]
HI @MPardey please bear with us and try to start the path with a forward slash and I hope its the 'full absolute path', relative paths are being handled poorly now.

e.g. nifti2dicom -d /example-notebooks/books/041/DICOM/PA000001/ST000001/SE000001/ -n example-notebooks/books/041/bids/derivatives/qsmxt-2024-11-19-235919/sub-041/ses-20241101/anat/sub-041_ses-20241101_run-01_acq-qsm_Chimap.nii -o /example-notebooks/books/041/QSM-dicomconvert/ -desc "QSM Analysis" -t img

**Comment by MPardey** [comment]
Morning, Thanks for the suggestion. I missed the first part of the path sorry. Please see below what I tried this morning. I included the permissions for this directory in case that is impacting something, although we have written a test .txt file to the output folder to check we can write to it.

(base) jovyan@neurodesktop-2024-10-22:~$ pwd
/home/jovyan
(base) jovyan@neurodesktop-2024-10-22:~$ ls -l
total 36
drwxr-sr-x 3 jovyan users 4096 Oct 28 06:22 Desktop
-rw-r--r-- 1 jovyan users 2559 Oct 31 03:34 MargQSMplay.txt
-rw-r--r-- 1 jovyan users  145 Nov 21 04:02 alsfsl
-rw-r--r-- 1 jovyan users  413 May 25  2024 conda-readme.md
drwxr-sr-x 2 jovyan users 4096 Oct 24 01:16 dicoms-sorted
drwxr-sr-x 6 jovyan users 4096 Nov 21 04:02 example-notebooks
drwxr-sr-x 3 jovyan users 4096 Nov 12 03:43 funtional_imaging
lrwxrwxrwx 1 jovyan users   22 Oct 16 09:02 neurodesktop-storage -> /neurodesktop-storage/
drwx------ 1 jovyan users    0 Nov 22 04:32 thinclient_drives
-rw-r--r-- 1 jovyan users  943 Oct 24 02:14 untitled.txt
drwsrwsr-x 2 jovyan users 4096 Oct 20  2023 work
(base) jovyan@neurodesktop-2024-10-22:~$ nifti2dicom -d /home/jovyan/example-notebooks/books/041/DICOM/PA000001/ST000001/SE000001/ -n /home/jovyan/example-notebooks/books/041/bids/derivatives/qsmxt-2024-11-19-235919/sub-041/ses-20241101/anat/sub-041_ses-20241101_run-01_acq-qsm_Chimap.nii -o /home/jovyan/example-notebooks/books/041/QSM-dicomconvert/ -desc "QSM Analysis" -t img
 
          _ ______  _ ___       ___                    
   ____  (_) __/ /_(_)__ \ ____/ (_)________  ____ ___ 
  / __ \/ / /_/ __/ /__/ // __  / / ___/ __ \/ __ `__ \
 / / / / / __/ /_/ // __// /_/ / / /__/ /_/ / / / / / /
/_/ /_/_/_/  \__/_//____/\__,_/_/\___/\____/_/ /_/ /_/
A package to convert NIfTI images to DICOM format using a reference DICOM series. Nifti2dicom is a part of the ENHANCE.PET (https://enhance.pet) framework
 
(base) jovyan@neurodesktop-2024-10-22:~$ 

Still no output and no errors.
Appreciate your assistance.

Cheers

**Comment by Keyn34** [comment]
This is highly odd. Even if you had no permission for the directories, you would certainly get an error or some message. Can you share the data? It would be way easier for me to tackle this issue by trying to replicate the problem. 

Thanks a lot

**Comment by luisa-sophia** [comment]
Hi, I am getting the same issue... no output, no erros, just the niftitodicom info header. I am feeding in a 4D nifti file. Tried it with absolute paths as well, and there is no difference, it just doesn't generate any output. This is my command:

ubuntu@luisa:/home/work/luisa/General_Code/MRE/nifti_to_dicom$ nifti2dicom -d ./ref_dicoms -n ./sub-neuromod001elastasfmri_task-resting_acq-MREmagnitude_bold.nii -o ./out_dicoms -desc "test" -t img -v sms

File structure:
![Image](https://github.com/user-attachments/assets/d042419b-8cb0-4cbd-bf20-0f60b7195061)

Note: I have just copied a single dicom file as reference file, might be that it is not enough? Also tried with passing the json sidecar file as well - no difference

Thanks for your help!
Luisa

**Comment by Keyn34** [comment]
Sorry for your trouble, guys. 

It's good that you tried absolute paths. Let's try using always absolute paths so that we can rule out this one as well. 
I am unsure if 4D is taken in well, especially when you have only one DICOM file. Did you also try doing a 3D image with the corresponding number of DICOM files? 

Thanks so much.

**Comment by LalithShiyam** [comment]
Yep; we need 4d dicom for 4d nifti… Sent from my iPhoneOn Jan 22, 2025, at 3:10 PM, Sebastian Gutschmayer ***@***.***> wrote:﻿
Sorry for your trouble, guys.
It's good that you tried absolute paths. Let's try using always absolute paths so that we can rule out this one as well.
I am unsure if 4D is taken in well, especially when you have only one DICOM file. Did you also try doing a 3D image with the corresponding number of DICOM files?
Thanks so much.

**Comment by Keyn34** [comment]
@luisa-sophia make sure that the directory you want to write the DICOM files to does not exist yet. It should work then. 

I will close this issue. Please feel free to let me know if this is still relevant to you.
